Fix for +fido2 variant compilation for 8.4p1.

Addresses https://trac.macports.org/ticket/62890.

The first part of this is a commit picked from upstream making sure that the
correct variant of the "sha2.h" file is being included.

The second part seemed to be necessary for exactly the same reason in
sk-usbhid.c although it was not applied upstream and doesn't seem to need to be.
I simply remove the reference to <sha2.h> because the openbsd-compat version is
already referenced in "includes.h".



From 86cc8ce002ea10e88a4c5d622a8fdfab8a7d261f Mon Sep 17 00:00:00 2001
From: Damien Miller <djm@mindrot.org>
Date: Sat, 3 Oct 2020 13:38:55 +1000
Subject: [PATCH] use relative rather than system include here

---
 openbsd-compat/sha2.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/openbsd-compat/sha2.c b/openbsd-compat/sha2.c
index e36cc24e..ce936e26 100644
--- a/openbsd-compat/sha2.c
+++ b/openbsd-compat/sha2.c
@@ -45,7 +45,7 @@
 #define MAKE_CLONE(x, y)       void __ssh_compat_make_clone_##x_##y(void)
 
 #include <string.h>
-#include <sha2.h>
+#include "openbsd-compat/sha2.h"
 
 /*
  * UNROLLED TRANSFORM LOOP NOTE:
-- 




--- a/sk-usbhid.c
+++ b/sk-usbhid.c
@@ -26,9 +26,6 @@
 #include <stdio.h>
 #include <stddef.h>
 #include <stdarg.h>
-#ifdef HAVE_SHA2_H
-#include <sha2.h>
-#endif
 
 #ifdef WITH_OPENSSL
 #include <openssl/opensslv.h>
